src/shared/sleep-config.c

```diff
@@ -526,6 +528,71 @@ int get_total_suspend_interval(Hashmap *last_capacity, usec_t *ret) {
         return 0;
 }
 
+/* Return true if all batteries have acpi_btp support */
+int battery_trip_point_alarm_exists(void) {
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        sd_device *dev;
+        int r, battery_count = 0, battery_alarm_count =0;
+
+        r = battery_enumerator_new(&e);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to initialize battery enumerator: %m");
+
+        FOREACH_DEVICE(e, dev) {
+                const char *s;
+                int battery_alarm;
+
+                r = sd_device_get_sysattr_value(dev, "alarm", &s);
+                if (r < 0) {
+                        log_device_debug_errno(dev, r, "Failed to read battery name, ignoring: %m");
```

Comment:
these shouldn't be ignored, same below - unless it can be guaranteed that alarms are supported on all detected batteries, the fallback should be used instead